File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.cs
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     public int val;
 *     public ListNode next;
 *     public ListNode(int val=0, ListNode next=null) {
 *         this.val = val;
 *         this.next = next;
 *     }
 * }
 */
public class Solution {
    public ListNode ModifiedList(int[] nums, ListNode head) {
        HashSet<int> set = new(nums);
        List<ListNode> helper = new List<ListNode>();

        while (head != null) {
            if (!set.Contains(head.val)) {
                helper.Add(head);
            }
            head = head.next;
        }
        helper.Reverse();
        ListNode prev = null;

        foreach (ListNode node in helper) {
            head = new ListNode(node.val, prev);
            prev = head;
        }

        return head;
    }
}

File: 3501-delete-nodes-from-linked-list-present-in-array/delete-nodes-from-linked-list-present-in-array.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def modifiedList(self, nums: List[int], head: Optional[ListNode]) -> Optional[ListNode]:
        dummy = ListNode(-1)
        t = dummy
        s = set(nums)

        while head:
            if head.val not in s:
                t.next = head
                t = t.next
            head = head.next

        t.next = None

        return dummy.next
